src/sagas/pushNotification.js

```diff
@@ -0,0 +1,432 @@
+import {
+  types,
+  pushRegisterSuccess,
+  pushRegisterFailed,
+  pushUpdateSuccess,
+  pushUpdateFailed,
+  pushUpdateDeviceId,
+  pushRegistrationRequested,
+  setIsShowingPinScreen,
+  pushLoadWalletRequested,
+  pushLoadWalletSuccess,
+  pushLoadWalletFailed,
+  pushInit,
+} from '../actions';
+import {
+  pushNotificationKey,
+  STORE,
+  WALLET_SERVICE_MAINNET_BASE_WS_URL,
+  WALLET_SERVICE_MAINNET_BASE_URL,
+} from '../constants';
+import NavigationService from '../NavigationService';
+import { getPushNotificationSettings } from '../utils';
+
+export const PUSH_API_STATUS = {
+  READY: 'ready',
+  FAILED: 'failed',
+  LOADING: 'loading',
+};
+
+/**
+ * this is the network name that will be used to load the wallet on the wallet-service,
+ * it is first hardcoded in the `startWallet` saga function, @see src\sagas\wallet.js.
+ */
+const NETWORK = 'mainnet';
+/**
+ * this is the message key for localization of new transaction when show amount is enabled
+ */
+const NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS = 'new_transaction_received_description_with_tokens';
+/**
+ * this is the message key for localization of new transaction when show amount is disabled
+ */
+const NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS = 'new_transaction_received_description_without_tokens';
+/**
+ * this is the message key for localization of new transaction title
+ */
+const NEW_TRANSACTION_RECEIVED_TITLE = 'new_transaction_received_title';
+/**
+ * this is the channel id for the transaction notification
+ */
+const TRANSACTION_CHANNEL_ID = 'transaction';
+const TRANSACTION_CHANNEL_NAME = t`Transaction`;
+
+const getDeviceId = async () => {
+  try {
+    const deviceId = await messaging().getToken();
+    return deviceId;
+  } catch (error) {
+    console.error(`Error getting deviceId: ${error.message}`, error);
+    return null;
+  }
+};
+
+/**
+ * localization utils to map the message key to the correct message to localize
+ */
+const localization = {
+  keys: new Set([
+    NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS,
+    NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS,
+    NEW_TRANSACTION_RECEIVED_TITLE
+  ]),
+  hasKey: (key) => localization.keys.has(key),
+  getMessage: (key, args) => {
+    if (!localization.hasKey(key)) {
+      console.log('Unknown localization key for push notification message.', key);
+      return '';
+    }
+
+    let message = '';
+    if (key === NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS) {
+      if (!args) {
+        console.log(`The args for push notification message key ${NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS} cannot be null or undefined.`, key);
+        return '';
+      }
+      const countArgs = args.length;
+      if (countArgs === 3) {
+        const [firstToken, secondToken, other] = args;
+        const otherCount = parseInt(other, 10);
+        message = t`You have received ${firstToken}, ${secondToken} and ${otherCount} other token on a new transaction.`;
```

Comment:
I didn't understand this message. Isn't it suppose to show the amount received? Maybe an example of the expected message would help.